atomicapp/nulecule/lib.py

```diff
@@ -98,12 +100,32 @@ def get_provider(self, provider_key=None, dry=False):
             tuple: (provider key, provider instance)
         """
         if provider_key is None:
-            provider_key = self.config.get('general', {}).get(
-                'provider', DEFAULT_PROVIDER)
+            if self.is_openshift:
+                provider_key = "openshift"
+            else:
+                provider_key = self.config.get('general', {}).get(
+                    'provider', DEFAULT_PROVIDER)
         provider_class = plugin.getProvider(provider_key)
         return provider_key, provider_class(
             self.get_context(), self.basepath, dry)
 
+    @property
+    def is_openshift(self):
+        """
+        The KUBERNETES_SERVICE_HOST env var should only exist
+        on an openshift or kubernetes environment.
+        Here we check if the "kubernetes" host has an openshift
+        API endpoint. If so, we're calling openshift.
+        """
+        if os.getenv("KUBERNETES_SERVICE_HOST"):
+            curl_cmd = ("curl -ks -w '%{http_code}' "
```

Comment:
I agree that would be better. I was concerned about adding another python dependency.